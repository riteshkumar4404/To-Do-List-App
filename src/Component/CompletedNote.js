import React, { Component } from "react";
import axios from "axios";
import Button from "react-bootstrap/Button";
import { Form } from "react-bootstrap";
import AxiosCall from "./AxiosCall";
import AddNote from "../Component/AddNote";
import { Navbar, Nav, NavDropdown } from "react-bootstrap";

class CompletedNote extends Component {
  constructor(props) {
    super(props);

    this.state = {
      notes: [],
      noteId: 0,
      currentTime: "",
      activeNote: 0,
      completed: ""
    };
  }

  getDate = () => {
    var currentTime = new Date();
    var currentTime =
      currentTime.getFullYear() +
      "-" +
      (currentTime.getMonth() + 1) +
      "-" +
      currentTime.getDate() +
      " " +
      currentTime.getHours() +
      ":" +
      currentTime.getMinutes() +
      ":" +
      currentTime.getSeconds();
    this.setState({ currentTime });
  };

  handleComplete(Noteid) {
    this.getDate();

    const user = {
      completeStatus: 0,
      noteId: Noteid,
      completed: null
    };

    AxiosCall.MarkComplete(user)
      .then(response => {
        if (response.ok) {
          console.log("Completed Successfully!");
          return;
        }
        throw new Error("Network response was not ok!");
      })
      .then(window.location.reload(true))
      .catch(e => {
        console.log(e);
      });
  }

  componentDidMount() {
    this.getDate();
    axios
      .get("http://localhost:9090/demorest/webapi/notes/note/completed")
      .then(res => {
        this.setState({ notes: res.data });
      });
  }
  render() {
    const { notes } = this.state;
    const allNotes = notes.map((note, index) => (
      <div
        key={index}
        className="card text-white bg-secondary mb-3"
        style={{ margin: "30px", textAlign: "left", maxWidth: "40rem" }}
      >
        <div className="card-header">Note</div>
        <div className="card-body">
          <h4 className="card-title">Note {index + 1}</h4>
          <p className="card-text">
            {note.note}
            <br />
            Note Id: {note.noteId}
            <br />
            Notification time : {note.notificationTime}
          </p>

          <Form inline>
            <Button
              variant="light"
              className="btn btn-outline-primary mr-1"
              onClick={() => {
                var result = window.confirm(
                  "Task will be marked uncompleted ?"
                );
                if (result) {
                  this.handleComplete(note.noteId);
                }
              }}
            >
              Mark as uncomplete
            </Button>
          </Form>
        </div>
      </div>
    ));
    const noNotes = (
      <div>
        <h4>No notes available!</h4>
      </div>
    );
    return (
      <div>
        <div className="App">
          <Navbar bg="light" expand="lg">
            <Navbar.Brand href="#home">
              <h2>To DO List</h2>
            </Navbar.Brand>
            <Navbar.Toggle aria-controls="basic-navbar-nav" />
            <Navbar.Collapse id="basic-navbar-nav">
              <Nav className="mr-auto">
                <Nav.Link onClick={() => (window.location.href = "/")}>
                  Home
                </Nav.Link>
                <Nav.Link onClick={() => (window.location.href = "/completed")}>
                  Completed Note
                </Nav.Link>
              </Nav>
              <Form inline>
                <AddNote />
              </Form>
            </Navbar.Collapse>
          </Navbar>
          <div style={{ margin: "auto", width: "50%", padding: "10px" }}>
            {notes.length > 0 ? allNotes : noNotes}
          </div>
        </div>
      </div>
    );
  }
}

export default CompletedNote;
