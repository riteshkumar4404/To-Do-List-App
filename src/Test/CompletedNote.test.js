import React from "react";
import CompletedNote from "../Component/CompletedNote";
import { configure } from "enzyme";
import { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
const event = {
  target: { name: "test", value: "server" },

  preventDefault: () => {}
};
const target = { name: "riskType", value: "LOWRISK" };
const stateData = {
  notes: [],
  noteId: 1,
  currentTime: "12-03-2020",
  activeNote: 0,
  completed: "12-03-2020"
};

configure({ adapter: new Adapter() });

describe("CompletedNoteNote", () => {
  it("should render without error", () => {
    const wrapper = shallow(<CompletedNote />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("CompletedNoteNote testing", () => {
  it("should render a AddNote and trigger handleChange function", () => {
    const CompletedNoteNoteFormView = shallow(<CompletedNote />);
    const inputField = CompletedNoteNoteFormView.find("Form");
    const mockEvent = {
      target: {
        value: "test"
      }
    };
    expect(CompletedNoteNoteFormView.first().shallow()).toMatchSnapshot();
  });
});

describe("CompletedNoteNote testing2", () => {
  it("it should render without any error ", () => {
    const wrapper = shallow(<CompletedNote />);

    const wrapperFunction = wrapper.instance();
    wrapperFunction.handleComplete();
    wrapperFunction.componentDidMount();

    expect(wrapper).toMatchSnapshot();
  });
});
