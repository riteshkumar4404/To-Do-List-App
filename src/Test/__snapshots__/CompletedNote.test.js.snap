// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompletedNoteNote should render without error 1`] = `ShallowWrapper {}`;

exports[`CompletedNoteNote testing should render a AddNote and trigger handleChange function 1`] = `ShallowWrapper {}`;

exports[`CompletedNoteNote testing2 it should render without any error  1`] = `ShallowWrapper {}`;
